extension/background.js

```diff
@@ -492,9 +492,35 @@ function loadSurvey(element, decision, timePromptShown, timePromptClicked) {
   });
 }
 
-// Trigger the new survey prompt when the participant makes a decision about an
-// experience sampling element.
+/**
+ * Record basic information about the event.
+ * @param {object} element The browser element of interest.
+ * @param {object} decision The decision the participant made.
+ */
+function recordEvent(element, decision) {
+  var responses = [];
+  responses.push(new SurveySubmission.Response(
+      'Full event type', element['name']));
+  responses.push(new SurveySubmission.Response(
+      'Response', decision['name']));
+  responses.push(new SurveySubmission.Response(
+      'Details', decision['details']));
+  responses.push(new SurveySubmission.Response(
+      'Learn more', decision['learn_more']));
+  getParticipantId().then(function(participantId) {
+    var record = new SurveySubmission.SurveyRecord(
+      constants.FindEventType(element['name']),
+      participantId,
+      (new Date),
+      responses);
+    SurveySubmission.saveSurveyRecord(record);
```

Comment:
Well, they are labelled as "Full event type" etc. which is different from a
question. So this ought to be easily recognizable as an event and not a
question.

On Tue, Jan 13, 2015 at 12:34 AM, meacer <notifications@github.com> wrote:

> In extension/background.js
> <https://github.com/GoogleChrome/experience-sampling/pull/159#discussion-diff-22848653>
> :
>
> > +  var responses = [];
> > +  responses.push(new SurveySubmission.Response(
> > +      'Full event type', element['name']));
> > +  responses.push(new SurveySubmission.Response(
> > +      'Response', decision['name']));
> > +  responses.push(new SurveySubmission.Response(
> > +      'Details', decision['details']));
> > +  responses.push(new SurveySubmission.Response(
> > +      'Learn more', decision['learn_more']));
> > +  getParticipantId().then(function(participantId) {
> > +    var record = new SurveySubmission.SurveyRecord(
> > +      constants.FindEventType(element['name']),
> > +      participantId,
> > +      (new Date),
> > +      responses);
> > +    SurveySubmission.saveSurveyRecord(record);
>
> Hmm, how do you distinguish between these events and actual survey
> responses? There doesn't seem to be a flag to mark these as simple events.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleChrome/experience-sampling/pull/159/files#r22848653>
> .
>